deployment/config.go

```diff
@@ -139,6 +139,24 @@ func (t TerraformMap) String() string {
 	return "{" + strings.Join(pairs, ", ") + "}"
 }
 
+// ClusterSubnetIDs contains the subnet ids for the different types of instances.
+type ClusterSubnetIDs struct {
+	App           []string `json:"app"`
+	Job           []string `json:"job"`
+	Proxy         []string `json:"proxy"`
+	Agent         []string `json:"agent"`
+	ElasticSearch []string `json:"elasticsearch"`
+	Metrics       []string `json:"metrics"`
+	Keycloak      []string `json:"keycloak"`
+	Database      []string `json:"database"`
+	Redis         []string `json:"redis"`
+}
+
+// IsAnySet returns true if any of the subnet ids are set.
+func (c *ClusterSubnetIDs) IsAnySet() bool {
+	return len(c.App) > 0 || len(c.Job) > 0 || len(c.Proxy) > 0 || len(c.Agent) > 0 || len(c.ElasticSearch) > 0 || len(c.Metrics) > 0 || len(c.Keycloak) > 0 || len(c.Database) > 0 || len(c.Redis) > 0
+}
```

Comment:
I think that using something like `reflect.DeepEqual` to avoid every single field and automatically support future ones may be a worthwhile simplification given performance is not really a concern. Not a huge deal though.